File: app/src/main/java/com/example/vlad/financemanager/ui/adapters/SimpleSpinnerAdapter.java
package com.example.vlad.financemanager.ui.adapters;

import android.content.Context;
import android.support.annotation.NonNull;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import com.example.vlad.financemanager.R;
import com.example.vlad.financemanager.data.models.SpinnerItem;

import java.util.List;

public class SimpleSpinnerAdapter extends ArrayAdapter<String> {

    private int dropDownTextColor;
    private int selectedTextColor;

    public SimpleSpinnerAdapter(Context context, int resource, String[] spinnerItems, int selectedTextColor, int dropDownTextColor) {
        super(context, resource, spinnerItems);
        this.selectedTextColor = selectedTextColor;
        this.dropDownTextColor = dropDownTextColor;
    }

    @Override
    public View getDropDownView(int position, View convertView, @NonNull ViewGroup parent) {
        return initView(position, convertView, parent, dropDownTextColor);
    }

    @NonNull
    @Override
    public View getView(int position, View convertView, @NonNull ViewGroup parent) {
        View view = initView(position, convertView, parent, selectedTextColor);
        return view;
    }

    private View initView(int position, View convertView, ViewGroup parent, int textColor) {
        if (convertView == null) {
            convertView = LayoutInflater.from(getContext()).inflate(
                    R.layout.simple_spinner_item, parent, false);
        }

        TextView accountName = convertView.findViewById(R.id.accountTypeTextView);
        accountName.setText(getItem(position));
        accountName.setTextColor(textColor);

        return convertView;
    }
}

File: app/src/main/java/com/example/vlad/financemanager/ui/adapters/SimpleSpinnerAdapter.kt
package com.example.vlad.financemanager.ui.adapters

import android.content.Context
import android.support.annotation.ColorRes
import android.support.v4.content.ContextCompat
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView

import com.example.vlad.financemanager.R

class SimpleSpinnerAdapter(context: Context, resource: Int, spinnerItems: Array<String>,
                           @param:ColorRes @field:ColorRes private val selectedTextColor: Int,
                           @param:ColorRes @field:ColorRes private val dropDownTextColor: Int) :
        ArrayAdapter<String>(context, resource, spinnerItems) {

    override fun getDropDownView(position: Int, convertView: View?, parent: ViewGroup): View {
        return initView(position, convertView, parent, dropDownTextColor)
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        return initView(position, convertView, parent, selectedTextColor)
    }

    private fun initView(position: Int, convertView: View?, parent: ViewGroup, textColor: Int): View {
        val view = convertView ?: LayoutInflater.from(context).inflate(
                R.layout.simple_spinner_item, parent, false)

        val accountName = view.findViewById<TextView>(R.id.accountTypeTextView)
        accountName.text = getItem(position)
        accountName.setTextColor(ContextCompat.getColor(context, textColor))

        return view
    }
}
